chrome/content/zotero/xpcom/attachments.js

```diff
@@ -1010,21 +1010,21 @@ Zotero.Attachments = new function () {
 		let collections = options.collections;
 		let title = options.title;
 		let saveOptions = options.saveOptions;
-		
-		if (!title) {
-			// TODO Better attachment name
-			title = Zotero.getString('itemFields.attachmentPDF');
-		}
 
 		if (parentItemID && collections) {
 			throw new Error("parentItemID and collections cannot both be provided");
 		}
 		
+		// Create a temporary file
+		let parentItem = Zotero.Items.get(parentItemID);
+		let fileBaseName = this.getFileBaseNameFromItem(parentItem, { attachmentTitle: title });
+		let ext = this._getExtensionFromURL(url, contentType);
+		let filename = Zotero.File.truncateFileName(`${fileBaseName}.${ext}`, 100);
```

Comment:
I don't think we need to truncate. We only do that currently when the filename comes from the URL, to handle cases where the last path component is very long. Here we're doing standard metadata-based renaming, and the renaming format already handles title truncation.

I can't remember if we handle filesystem errors from too-long filenames, but that would be handled generally, not when assigning the name.

Reply:
And if `_getExtensionFromURL` is a concern, we should just add reasonable truncation (20 characters?) to that.

Reply:
`truncateFileName` already treats long extensions (20 chars) as part of filename. And in my testing this truncation often does get triggered, and also mimics previous behaviour, and we probably don't want to produce super-long filenames.

Reply:
Not sure what you mean here.

We use `truncateFileName` in two places: `importFromDocument` and `importFromSnapshotContent`. And in both cases we're using it because the "filename" is just coming from the URL. We don't use it for renamed-attachment saving, so we shouldn't be using it here.

I'm saying that 1) if `_getExtensionFromURL` can result in a very long string (because there's no known Content-Type and the URL has a period followed by lot of characters?), we should just handle that there and 2) if there's some problem with overly long filenames being saved for renamed attachments, it should be handled centrally where it applies to all uses of `getFileBaseNameFromItem`, not here. (But the default rename string already truncates titles to 100 characters.)

Reply:
Ah, you're right. I got mixed up when working on this. There are a lot of renaming methods here and every attachment import source does something slightly different!

I didn't identify any specific issues, was just following what we do in other places, so I don't think there's any need to do something special here, since importFromURL which was used previously to save attachment files didn't truncate either.